# Hello DBMS+ : De SQL aux Empreintes Carbone
Ce notebook présente un parcours complet d'analyse de données, alliant **requêtes SQL**, **exploration de tables multiples**, et **calcul de l’empreinte carbone**.  
L'objectif est de démontrer comment exploiter différentes sources de données pour générer des **insights pertinents**, à la fois pour la compréhension des données mondiales et pour l'évaluation environnementale.

Le notebook est structuré en deux grandes parties :  
1. **Jobs SQL 1 à 9** : exploration, filtrage, agrégation et jointures sur plusieurs tables (`world`, `students`, `nobel`, `SomeCompany`, etc.), permettant de comprendre et manipuler les données de manière progressive et pédagogique.  
2. **Big Job : Empreinte Carbone** : analyse des données de consommation énergétique par pays et région, calcul des émissions de CO2, visualisations et estimation du nombre d’arbres nécessaires pour compenser ces émissions.

In [1]:
# Imports

# Manipulation et analyse de données
import pandas as pd
import numpy as np

# Visualisation
import matplotlib.pyplot as plt
import seaborn as sns

# Pour afficher les graphiques directement dans le notebook
%matplotlib inline

# SQL magic pour exécuter des requêtes directement dans les cellules
%load_ext sql


# formatage des DataFrames pour la lisibilité
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', 100)
pd.set_option('display.float_format', '{:.2f}'.format)

## Exploration et Analyse de Données avec SQL

In [2]:
# connecteur Postgresql
%sql postgresql://postgres:5219--ZmId*@localhost:65432/postgres
# %sql postgresql://postgres:****--****@localhost:*****/postgres

## Big Job

### Nettoyage du fichier csv 

#### Notes : 

1. **CSV et format**  
   - Encodage spécifique : `WIN1252`  
   - Délimiteur : `;`  
   - Ligne 143 complètement vide → problème d’import dans PostgreSQL  

2. **Séparation du dataset en deux tables**  
   - **Table `country`** → toutes les lignes avant `"World"` (noms de pays)  
   - **Table `world`** → ligne `"World"` + toutes les lignes régionales  

3. **Minuscule pour les noms**  
   - Tous les noms de pays sont stockés en minuscules (`LOWER(country)`) pour faciliter les futures requêtes  

4. **Colonnes des tables**  
   - `country` : colonnes → `country, coal, gas, oil, hydro, renewable, nuclear`  
   - `world` : colonnes → `region, coal, gas, oil, hydro, renewable, nuclear`


In [3]:
# import des données du CSV dans un dataframe pandas
csv_path= r"C:\Users\Paul-Emmanuel Buffe\Desktop\la_plateforme\travaux_la_plateforme\hello-dbms\data\carbon-footprint-data.csv"
df =pd.read_csv(csv_path,delimiter=';', encoding= 'windows-1252') # Encodage spécifique

# Suppression des lignes vides
df.dropna(how='all', inplace=True) # on ne supprime que les lignes qui sont complètement vides

In [4]:
print(df.head(3))

   Country  Coal   Gas   Oil  Hydro  Renewable  Nuclear
0  Albania  0.00  0.00  0.00 100.00       0.00     0.00
1  Algeria  0.00 97.80  1.80   0.40       0.00     0.00
2   Angola  0.00  0.00 46.80  53.20       0.00     0.00


### Création des tables
**Table "original_raw"**

In [5]:
# %%sql
# CREATE TABLE IF NOT EXISTS original_raw
#     country VARCHAR(100),
#     coal NUMERIC NULL,
#     gas NUMERIC NULL,
#     oil NUMERIC NULL,
#     hydro NUMERIC NULL,
#     renewable NUMERIC NULL,
#     nuclear NUMERIC NULL
# );

In [6]:
# insertion des données depuis le df
# for i, row in df.iterrows():
#     sql= f"""
#             INSERT INTO original_raw (country, coal, gas, oil, hydro, renewable, nuclear)
#             VALUES ('{row['Country']}', '{row['Coal']}', '{row['Gas']}', '{row['Oil']}', '{row['Hydro']}', '{row['Renewable']}', '{row['Nuclear']}')
#           """
#     %sql {sql}

**Table "country"**

In [7]:
# %%sql
# CREATE TABLE IF NOT EXISTS country(
#     country VARCHAR(100),
#     coal NUMERIC NULL,
#     gas NUMERIC NULL,
#     oil NUMERIC NULL,
#     hydro NUMERIC NULL,
#     renewable NUMERIC NULL,
#     nuclear NUMERIC NULL
# );

In [8]:
# %%sql
# INSERT INTO country(country, coal, gas, oil, hydro, renewable, nuclear)
# SELECT 
#     LOWER(country), 
#     coal, 
#     gas, 
#     oil, 
#     hydro, 
#     renewable, 
#     nuclear
# FROM original_raw
# WHERE country NOT IN ('World', 'East Asia & Pacific', 'Europe & Central',
#                       'Latin America & Caribbean', 'Middle East & North Afrika',
#                       'North America', 'South Asia', 'Sub­Saharan Africa');

**Table "world"**

In [9]:
# %%sql
# CREATE TABLE IF NOT EXISTS world(
#     region VARCHAR(100),
#     coal NUMERIC NULL,
#     gas NUMERIC NULL,
#     oil NUMERIC NULL,
#     hydro NUMERIC NULL,
#     renewable NUMERIC NULL,
#     nuclear NUMERIC NULL
# );

In [10]:
# %%sql
# INSERT INTO world(region, coal, gas, oil, hydro, renewable, nuclear)
# SELECT LOWER(country), 
#         coal, 
#         gas, 
#         oil, 
#         hydro, 
#         renewable, 
#         nuclear
# FROM original_raw
# WHERE country IN ('World', 'East Asia & Pacific', 'Europe & Central',
#                       'Latin America & Caribbean', 'Middle East & North Afrika',
#                       'North America', 'South Asia', 'Sub­Saharan Africa');

**Vérification des tables crées**

In [11]:
%sql SELECT * FROM country LIMIT 3;

 * postgresql://postgres:***@localhost:65432/postgres
3 rows affected.


[('albania', Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0'), Decimal('0.0'), Decimal('0.0')),
 ('algeria', Decimal('0.0'), Decimal('97.8'), Decimal('1.8'), Decimal('0.4'), Decimal('0.0'), Decimal('0.0')),
 ('angola', Decimal('0.0'), Decimal('0.0'), Decimal('46.8'), Decimal('53.2'), Decimal('0.0'), Decimal('0.0'))]

In [12]:
%sql SELECT * FROM world LIMIT 3;

 * postgresql://postgres:***@localhost:65432/postgres
3 rows affected.


[('world', Decimal('40.7'), Decimal('21.6'), Decimal('4.1'), Decimal('16.2'), Decimal('6.0'), Decimal('10.6')),
 ('east asia & pacific', Decimal('60.6'), Decimal('13.5'), Decimal('2.2'), Decimal('15.0'), Decimal('4.2'), Decimal('3.8')),
 ('europe & central', Decimal('24.1'), Decimal('24.3'), Decimal('1.3'), Decimal('16.6'), Decimal('10.5'), Decimal('22.4'))]

### Préparation et analyse des sources d’énergie des pays pour l’étude des émissions de CO2

**Analyse des données au niveau mondial**

In [13]:
%%sql
WITH filtered AS (
  SELECT *
  FROM world
  WHERE region = 'world'
)
SELECT 'coal' AS source, coal AS value FROM filtered
UNION ALL
SELECT 'gas' AS source, gas AS value FROM filtered
UNION ALL
SELECT 'oil' AS source, oil AS value FROM filtered
UNION ALL
SELECT 'hydro' AS source, hydro AS value FROM filtered
UNION ALL
SELECT 'renewable' AS source, renewable AS value FROM filtered
UNION ALL
SELECT 'nuclear' AS source, nuclear AS value FROM filtered;




 * postgresql://postgres:***@localhost:65432/postgres
6 rows affected.


[('coal', Decimal('40.7')),
 ('gas', Decimal('21.6')),
 ('oil', Decimal('4.1')),
 ('hydro', Decimal('16.2')),
 ('renewable', Decimal('6.0')),
 ('nuclear', Decimal('10.6'))]

**Analyse des différents mix energétique des grands ensembles régionaux : Energies Fossiles VS Autres Energies**

In [14]:
%%sql
SELECT
    region,
    CEIL(gas + oil + coal) AS fossil_energie,
    ROUND(hydro + renewable + nuclear) AS others
FROM world WHERE region = 'world'
UNION ALL
SELECT*
FROM (SELECT 
region,
    (gas + oil + coal) AS fossil_energie,
    ROUND(hydro + renewable + nuclear) AS others
FROM world
ORDER BY 2 DESC LIMIT 1)
UNION ALL
SELECT*
FROM (SELECT 
region,
    (gas + oil + coal) AS fossil_energie,
    ROUND(hydro + renewable + nuclear) AS others
FROM world
ORDER BY 2 asc LIMIT 1)

 * postgresql://postgres:***@localhost:65432/postgres
3 rows affected.


[('world', Decimal('67'), Decimal('33')),
 ('middle east & north afrika', Decimal('96.3'), Decimal('3')),
 ('latin america & caribbean', Decimal('43.1'), Decimal('55'))]

**Régions du monde pour lesquelles le charbon rentre dans plus du quart de leur production d'electricité**

In [15]:
%%sql
SELECT 
    DISTINCT region,
    MAX(coal) AS max_coal
FROM world
WHERE 1=1
AND region NOT IN ('world')
AND coal > 25
group by 1
ORDER BY 2 DESC

 * postgresql://postgres:***@localhost:65432/postgres
4 rows affected.


[('south asia', Decimal('65.7')),
 ('east asia & pacific', Decimal('60.6')),
 ('sub\xadsaharan africa', Decimal('51.4')),
 ('north america', Decimal('35.7'))]

**Analyse de l'utilisation des énérgies dites "propres" par région.**

**Energies renouvelables:**

In [16]:
%%sql
SELECT 
    DISTINCT region,
    MAX(renewable) AS max_renewable
FROM world
WHERE region NOT IN ('world')
group by 1
ORDER BY 2 DESC
LIMIT 1

 * postgresql://postgres:***@localhost:65432/postgres
1 rows affected.


[('europe & central', Decimal('10.5'))]

**Energie Nucléaire**

In [17]:
%%sql
SELECT 
    DISTINCT region,
    MAX(nuclear) AS max_nuclear
FROM world
WHERE region NOT IN ('world')
group by 1
ORDER by 2 DESC
LIMIT 1

 * postgresql://postgres:***@localhost:65432/postgres
1 rows affected.


[('europe & central', Decimal('22.4'))]

**Energie Hydrolique**

In [18]:
%%sql
SELECT 
    region,
    MAX(hydro) AS max_hydro
FROM world
GROUP BY region
ORDER BY max_hydro DESC
LIMIT 1



 * postgresql://postgres:***@localhost:65432/postgres
1 rows affected.


[('latin america & caribbean', Decimal('46.5'))]

**Analyses des mix energétiques de la communanuté mondiale**

**Les pays avec plus de 66 % d'énergies dites propres**

In [19]:
%%sql
SELECT
country,
coal,
gas,
oil,
nuclear,
hydro,
renewable,
nuclear + hydro + renewable AS "total_energies_propres"
FROM country
WHERE nuclear + hydro + renewable >= 66
ORDER by 8 DESC



 * postgresql://postgres:***@localhost:65432/postgres
34 rows affected.


[('paraguay', Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0'), Decimal('0.0'), Decimal('100.0')),
 ('albania', Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0'), Decimal('0.0'), Decimal('100.0')),
 ('nepal', Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('99.8'), Decimal('0.2'), Decimal('100.0')),
 ('iceland', Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('71.0'), Decimal('28.9'), Decimal('99.9')),
 ('congo, dem. rep.', Decimal('0.0'), Decimal('0.1'), Decimal('0.0'), Decimal('0.0'), Decimal('99.9'), Decimal('0.0'), Decimal('99.9')),
 ('ethiopia', Decimal('0.0'), Decimal('0.0'), Decimal('0.1'), Decimal('0.0'), Decimal('95.6'), Decimal('4.3'), Decimal('99.9')),
 ('namibia', Decimal('0.0'), Decimal('0.0'), Decimal('0.9'), Decimal('0.0'), Decimal('99.1'), Decimal('0.0'), Decimal('99.1')),
 ('sweden', Decimal('0.6'), Decimal('0.3'), Decimal('0.2'), Decimal('42.3'), Decimal('41.5'), Decimal('14.3'), Decimal('98.1')),
 ('norway', Decimal('0.1'), Decimal('1.8'), Decimal('0.0'), Decimal('0.0'), Decimal('96.0'), Decimal('1.7'), Decimal('97.7')),
 ('switzerland', Decimal('0.0'), Decimal('0.7'), Decimal('0.1'), Decimal('39.3'), Decimal('54.3'), Decimal('3.8'), Decimal('97.4')),
 ('zambia', Decimal('0.0'), Decimal('0.0'), Decimal('2.8'), Decimal('0.0'), Decimal('97.2'), Decimal('0.0'), Decimal('97.2')),
 ('tajikistan', Decimal('0.0'), Decimal('2.9'), Decimal('0.0'), Decimal('0.0'), Decimal('97.1'), Decimal('0.0'), Decimal('97.1')),
 ('france', Decimal('2.2'), Decimal('2.3'), Decimal('0.3'), Decimal('78.4'), Decimal('11.3'), Decimal('5.1'), Decimal('94.8')),
 ('kyrgyz republic', Decimal('7.4'), Decimal('0.8'), Decimal('0.6'), Decimal('0.0'), Decimal('91.3'), Decimal('0.0'), Decimal('91.3')),
 ('mozambique', Decimal('0.0'), Decimal('8.8'), Decimal('0.0'), Decimal('0.0'), Decimal('91.2'), Decimal('0.0'), Decimal('91.2')),
 ('uruguay', Decimal('0.0'), Decimal('0.0'), Decimal('9.1'), Decimal('0.0'), Decimal('74.2'), Decimal('16.8'), Decimal('91.0')),
 ('costa rica', Decimal('0.0'), Decimal('0.0'), Decimal('10.2'), Decimal('0.0'), Decimal('65.7'), Decimal('24.0'), Decimal('89.7')),
 ('togo', Decimal('0.0'), Decimal('0.0'), Decimal('12.0'), Decimal('0.0'), Decimal('84.5'), Decimal('3.5'), Decimal('88.0')),
 ('kenya', Decimal('0.0'), Decimal('0.0'), Decimal('18.5'), Decimal('0.0'), Decimal('35.8'), Decimal('45.7'), Decimal('81.5')),
 ('austria', Decimal('8.0'), Decimal('8.8'), Decimal('1.0'), Decimal('0.0'), Decimal('66.6'), Decimal('14.6'), Decimal('81.2')),
 ('georgia', Decimal('0.0'), Decimal('19.6'), Decimal('0.0'), Decimal('0.0'), Decimal('80.4'), Decimal('0.0'), Decimal('80.4')),
 ('slovak republic', Decimal('12.4'), Decimal('6.0'), Decimal('1.1'), Decimal('57.1'), Decimal('15.5'), Decimal('7.4'), Decimal('80.0')),
 ('canada', Decimal('9.9'), Decimal('9.4'), Decimal('1.2'), Decimal('16.4'), Decimal('58.3'), Decimal('4.5'), Decimal('79.2')),
 ('new zealand', Decimal('4.5'), Decimal('16.3'), Decimal('0.0'), Decimal('0.0'), Decimal('55.9'), Decimal('23.2'), Decimal('79.1')),
 ('sudan', Decimal('0.0'), Decimal('0.0'), Decimal('21.7'), Decimal('0.0'), Decimal('78.3'), Decimal('0.0'), Decimal('78.3')),
 ('brazil', Decimal('4.5'), Decimal('13.7'), Decimal('6.0'), Decimal('2.6'), Decimal('63.2'), Decimal('9.9'), Decimal('75.7')),
 ('slovenia', Decimal('21.9'), Decimal('2.2'), Decimal('0.2'), Decimal('37.1'), Decimal('35.5'), Decimal('3.0'), Decimal('75.6')),
 ('cameroon', Decimal('0.0'), Decimal('12.9'), Decimal('12.8'), Decimal('0.0'), Decimal('73.2'), Decimal('1.0'), Decimal('74.2')),
 ('colombia', Decimal('10.2'), Decimal('15.3'), Decimal('0.2'), Decimal('0.0'), Decimal('71.1'), Decimal('3.1'), Decimal('74.2')),
 ('croatia', Decimal('17.6'), Decimal('7.5'), Decimal('1.0'), Decimal('0.0'), Decimal('67.0'), Decimal('6.9'), Decimal('73.9')),
 ('finland', Decimal('17.4'), Decimal('8.1'), Decimal('0.3'), Decimal('34.6'), Decimal('19.7'), 

**Les pays avec plus de 66 % d'énergies fossiles**

In [20]:
%%sql
SELECT
country,
coal,
gas,
oil,
nuclear,
hydro,
renewable,
coal + gas + oil AS "total_energies_fossiles"
FROM country
WHERE coal + gas + oil >= 66
ORDER by 8 DESC

 * postgresql://postgres:***@localhost:65432/postgres
69 rows affected.


[('yemen, rep.', Decimal('0.0'), Decimal('38.6'), Decimal('61.4'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0')),
 ('qatar', Decimal('0.0'), Decimal('100.0'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0')),
 ('trinidad and tobago', Decimal('0.0'), Decimal('99.8'), Decimal('0.2'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0')),
 ('saudi arabia', Decimal('0.0'), Decimal('51.2'), Decimal('48.8'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0')),
 ('bahrain', Decimal('0.0'), Decimal('100.0'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0')),
 ('kuwait', Decimal('0.0'), Decimal('33.7'), Decimal('66.3'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0')),
 ('libya', Decimal('0.0'), Decimal('53.7'), Decimal('46.3'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0')),
 ('oman', Decimal('0.0'), Decimal('97.4'), Decimal('2.6'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0')),
 ('turkmenistan', Decimal('0.0'), Decimal('100.0'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0')),
 ('botswana', Decimal('95.8'), Decimal('0.0'), Decimal('4.2'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0')),
 ('brunei darussalam', Decimal('0.0'), Decimal('99.0'), Decimal('1.0'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0'), Decimal('100.0')),
 ('hong kong sar, china', Decimal('76.2'), Decimal('23.0'), Decimal('0.6'), Decimal('0.0'), Decimal('0.0'), Decimal('0.2'), Decimal('99.8')),
 ('united arab', Decimal('0.0'), Decimal('98.4'), Decimal('1.3'), Decimal('0.0'), Decimal('0.0'), Decimal('0.3'), Decimal('99.7')),
 ('jordan', Decimal('0.0'), Decimal('7.1'), Decimal('92.5'), Decimal('0.0'), Decimal('0.3'), Decimal('0.0'), Decimal('99.6')),
 ('south sudan', Decimal('0.0'), Decimal('0.0'), Decimal('99.6'), Decimal('0.0'), Decimal('0.0'), Decimal('0.4'), Decimal('99.6')),
 ('algeria', Decimal('0.0'), Decimal('97.8'), Decimal('1.8'), Decimal('0.0'), Decimal('0.4'), Decimal('0.0'), Decimal('99.6')),
 ('eritrea', Decimal('0.0'), Decimal('0.0'), Decimal('99.5'), Decimal('0.0'), Decimal('0.0'), Decimal('0.5'), Decimal('99.5')),
 ('benin', Decimal('0.0'), Decimal('0.0'), Decimal('99.5'), Decimal('0.0'), Decimal('0.0'), Decimal('0.5'), Decimal('99.5')),
 ('niger', Decimal('71.6'), Decimal('0.0'), Decimal('27.8'), Decimal('0.0'), Decimal('0.0'), Decimal('0.6'), Decimal('99.4')),
 ('belarus', Decimal('0.1'), Decimal('98.0'), Decimal('1.1'), Decimal('0.0'), Decimal('0.3'), Decimal('0.4'), Decimal('99.2')),
 ('lebanon', Decimal('0.0'), Decimal('0.0'), Decimal('98.9'), Decimal('0.0'), Decimal('1.1'), Decimal('0.0'), Decimal('98.9')),
 ('bangladesh', Decimal('2.0'), Decimal('82.0'), Decimal('14.7'), Decimal('0.0'), Decimal('1.1'), Decimal('0.3'), Decimal('98.7')),
 ('israel', Decimal('49.6'), Decimal('48.4'), Decimal('0.5'), Decimal('0.0'), Decimal('0.0'), Decimal('1.5'), Decimal('98.5')),
 ('kosovo', Decimal('96.9'), Decimal('0.0'), Decimal('0.3'), Decimal('0.0'), Decimal('2.8'), Decimal('0.0'), Decimal('97.2')),
 ('singapore', Decimal('1.1'), Decimal('95.3'), Decimal('0.7'), Decimal('0.0'), Decimal('0.0'), Decimal('1.7'), Decimal('97.1')),
 ('mongolia', Decimal('92.3'), Decimal('0.0'), Decimal('4.5'), Decimal('0.0'), Decimal('0.0'), Decimal('3.2'), Decimal('96.8')),
 ('malta', Decimal('0.0'), Decimal('0.0'), Decimal('96.7'), Decimal('0.0'), Decimal('0.0'), Decimal('3.3'), Decimal('96.7')),
 ('curacao', Decimal('0.0'), Decimal('0.0'), Decimal('96.4'), Decimal('0.0'), Decimal('0.0'), Decimal('3.6'), Decimal('96.4')),
 ('cuba', Decimal('0.0'), Decimal('14.4'), Decimal('81.6'), Decimal('0.0'), Decimal('0.5'), Decimal('3.5'), Decimal('96.0')),
 ('tunisia', Decimal('0.0'), Decimal('94.2'), Decimal('1.8'), Decimal('0.0'), Decimal('0.3'), Decimal('2.8'), Decimal('96.0')),
 ('iraq', Decimal('0.0'), Decimal('21.9'), Decimal('73.7'), Decimal('0.0'), Decimal('4.3'), D

**Distribution mondiale des mix energétiques de pays**

In [26]:
%%sql
SELECT
*
FROM country
WHERE coal >= 66 
ORDER BY 2 DESC

 * postgresql://postgres:***@localhost:65432/postgres
13 rows affected.


[('kosovo', Decimal('96.9'), Decimal('0.0'), Decimal('0.3'), Decimal('2.8'), Decimal('0.0'), Decimal('0.0')),
 ('botswana', Decimal('95.8'), Decimal('0.0'), Decimal('4.2'), Decimal('0.0'), Decimal('0.0'), Decimal('0.0')),
 ('south africa', Decimal('93.0'), Decimal('0.0'), Decimal('0.1'), Decimal('0.4'), Decimal('1.0'), Decimal('5.5')),
 ('mongolia', Decimal('92.3'), Decimal('0.0'), Decimal('4.5'), Decimal('0.0'), Decimal('3.2'), Decimal('0.0')),
 ('estonia', Decimal('87.4'), Decimal('0.6'), Decimal('0.3'), Decimal('0.2'), Decimal('10.9'), Decimal('0.0')),
 ('poland', Decimal('83.0'), Decimal('3.4'), Decimal('1.0'), Decimal('1.4'), Decimal('11.1'), Decimal('0.0')),
 ('hong kong sar, china', Decimal('76.2'), Decimal('23.0'), Decimal('0.6'), Decimal('0.0'), Decimal('0.2'), Decimal('0.0')),
 ('india', Decimal('75.1'), Decimal('4.9'), Decimal('1.8'), Decimal('10.2'), Decimal('5.2'), Decimal('2.8')),
 ('china', Decimal('72.6'), Decimal('2.0'), Decimal('0.2'), Decimal('18.6'), Decimal('4.1'), Decimal('2.3')),
 ('kazakhstan', Decimal('71.9'), Decimal('19.2'), Decimal('1.0'), Decimal('7.9'), Decimal('0.0'), Decimal('0.0')),
 ('niger', Decimal('71.6'), Decimal('0.0'), Decimal('27.8'), Decimal('0.0'), Decimal('0.6'), Decimal('0.0')),
 ('macedonia, fyr', Decimal('69.5'), Decimal('3.6'), Decimal('2.8'), Decimal('22.5'), Decimal('1.6'), Decimal('0.0')),
 ('serbia', Decimal('66.3'), Decimal('0.7'), Decimal('0.0'), Decimal('32.9'), Decimal('0.1'), Decimal('0.0'))]

In [25]:
%%sql
SELECT
*
FROM country
WHERE renewable >= 33
ORDER BY 2 DESC

 * postgresql://postgres:***@localhost:65432/postgres
3 rows affected.


[('denmark', Decimal('34.4'), Decimal('6.5'), Decimal('1.0'), Decimal('0.0'), Decimal('55.8'), Decimal('0.0')),
 ('kenya', Decimal('0.0'), Decimal('0.0'), Decimal('18.5'), Decimal('35.8'), Decimal('45.7'), Decimal('0.0')),
 ('nicaragua', Decimal('0.0'), Decimal('0.0'), Decimal('46.1'), Decimal('8.9'), Decimal('45.0'), Decimal('0.0'))]

In [22]:
%%sql
SELECT
*
FROM country
WHERE nuclear >= 50

 * postgresql://postgres:***@localhost:65432/postgres
3 rows affected.


[('france', Decimal('2.2'), Decimal('2.3'), Decimal('0.3'), Decimal('11.3'), Decimal('5.1'), Decimal('78.4')),
 ('hungary', Decimal('20.8'), Decimal('14.4'), Decimal('0.2'), Decimal('1.0'), Decimal('9.7'), Decimal('53.3')),
 ('slovak republic', Decimal('12.4'), Decimal('6.0'), Decimal('1.1'), Decimal('15.5'), Decimal('7.4'), Decimal('57.1'))]

## Synthèse des observations sur la répartition mondiale des sources d’énergie électrique

### Vue d’ensemble mondiale

* Les énergies fossiles (pétrole, gaz, charbon) représentent 67 % de la production mondiale d’électricité.
* Les énergies dites propres (nucléaire, hydraulique, renouvelables) représentent seulement 33 %.
* Certaines régions atteignent des niveaux extrêmes : 96 % d’énergies fossiles pour seulement 4 % d’énergies propres.

### Analyse par régions

* L’énergie hydraulique constitue la principale composante des énergies propres dans la plupart des régions.
* L’Amérique latine est la région la moins dépendante des énergies fossiles (45 %), grâce à son importante production hydraulique.
* Trois régions utilisent encore le charbon à plus de 50 % dans leur mix électrique.
* En Amérique du Nord, le charbon représente encore 35 % de la production.
* L’Europe est la seule région ayant développé à la fois :

  * le nucléaire (22,4 %),
  * les énergies renouvelables (10,5 %).
    → Ensemble, ces deux sources ne couvrent néanmoins qu’un tiers du mix énergétique européen.

### Analyse par pays

* Grâce à l’hydroélectricité, 34 pays atteignent plus de 66 % d’énergies propres dans leur mix électrique.
* À l’opposé, 69 pays utilisent plus de 66 % d’énergies fossiles.
* Plus de 13 pays produisent plus des deux tiers de leur électricité à partir du charbon, notamment :

  * Chine : 73 %,
  * Inde : 75 %.
* Certains pays cumulant pourtant de fortes parts de renouvelables restent encore dépendants du charbon :

  * Exemple : Danemark, 56 % renouvelables mais 34 % charbon.
* Quelques exceptions notables :

  * France : 78 % d’électricité produite grâce au nucléaire.
  * Kenya : environ 90 % issus de l’hydraulique et des renouvelables.
